RevenueCatUI/Templates/Example1Template.swift

```diff
@@ -104,23 +130,55 @@ private struct Example1TemplateContent: View {
         }
     }
 
-    @ViewBuilder
     private var offerDetails: some View {
-        // Fix-me: this needs to handle other types of intro discounts
-        let text = self.data.package.storeProduct.introductoryDiscount == nil
-            ? self.data.localization.offerDetails
-            : self.data.localization.offerDetailsWithIntroOffer
+        let detailsWithIntroOffer = self.data.localization.offerDetailsWithIntroOffer
 
-        Text(verbatim: text)
+        func text() -> String {
+            if let detailsWithIntroOffer = detailsWithIntroOffer, self.isEligibleForIntro {
+                return detailsWithIntroOffer
+            } else {
+                return self.data.localization.offerDetails
+            }
+        }
+
+        return Text(verbatim: text())
+            // Hide until we've determined intro eligibility
+            // only if there is a custom intro offer string.
+            .withPendingData(self.needsToWaitForIntroEligibility(detailsWithIntroOffer != nil))
```

Comment:
I tried to reduce as much duplication as possible between the 2 texts. This will be refactored further once we take it out of this template to be used by all.